darts/tests/explainability/test_shap_explainer.py

```diff
@@ -0,0 +1,348 @@
+
+    past_cov_ts = TimeSeries.from_times_and_values(
+        days_past_cov,
+        np.concatenate(
+            [
+                past_cov_1.reshape(-1, 1),
+                past_cov_2.reshape(-1, 1),
+                past_cov_3.reshape(-1, 1),
+            ],
+            axis=1,
+        ),
+    )
+
+    fut_cov_ts = TimeSeries.from_times_and_values(
+        days,
+        np.concatenate([fut_cov_1.reshape(-1, 1), fut_cov_2.reshape(-1, 1)], axis=1),
+    )
+
+    models = []
+    models.append(
+        LightGBMModel(
+            lags=4,
+            lags_past_covariates=[-1, -2, -3],
+            lags_future_covariates=[0],
+            output_chunk_length=4,
+            add_encoders=add_encoders,
+        )
+    )
+
+    models.append(
+        CatBoostModel(
+            lags=4,
+            lags_past_covariates=[-1, -2, -6],
+            lags_future_covariates=[0],
+            output_chunk_length=4,
+        )
+    )
+    models.append(
+        LinearRegressionModel(
+            lags=1,
+            lags_past_covariates=[-1, -2, -3],
+            lags_future_covariates=[0],
+            output_chunk_length=2,
+        )
+    )
+
+    models.append(
+        RegressionModel(
+            lags=4,
+            lags_past_covariates=[-1, -2, -3],
+            lags_future_covariates=[0],
+            output_chunk_length=2,
+            model=sklearn.tree.ExtraTreeRegressor(),
+        )
+    )
+
+    def test_creation(self):
+
+        # Model should be fitted first
+        with self.assertRaises(ValueError):
+            ShapExplainer(
+                self.models[0], self.target_ts, self.past_cov_ts, self.fut_cov_ts
+            )
+
+        # Model should be a RegressionModel
+        m = ExponentialSmoothing()
+        m.fit(self.target_ts["price"])
+        with self.assertRaises(ValueError):
+            ShapExplainer(m)
+
+        m = self.models[0].fit(
+            series=self.target_ts,
+            past_covariates=self.past_cov_ts,
+            future_covariates=self.fut_cov_ts,
+        )
+
+        # SHould have the same number of target, past and futures in the respective lists
+        with self.assertRaises(ValueError):
+            ShapExplainer(
+                self.models[0],
+                [self.target_ts, self.target_ts],
+                self.past_cov_ts,
+                self.fut_cov_ts,
+            )
+
+        # Missing a future covariate if you choose to use a new background
+        with self.assertRaises(ValueError):
+            ShapExplainer(
+                m, self.target_ts, background_past_covariates=self.past_cov_ts
+            )
+
+        # Missing a past covariate if you choose to use a new background
+        with self.assertRaises(ValueError):
+            ShapExplainer(
+                m, self.target_ts, background_future_covariates=self.fut_cov_ts
+            )
+
+        # good type of explainers
+        ShapExplain = ShapExplainer(m)
```

Comment:
variable names should be lowercase and seperated by "_".
Could you also adapt the same cases further down?
```suggestion
        shap_explain = ShapExplainer(m)
```